src/librustc_typeck/check/_match.rs

```diff
@@ -626,7 +626,7 @@ https://doc.rust-lang.org/reference/types.html#trait-objects");
 
         // Typecheck the patterns first, so that we get types for all the
         // bindings.
-        let all_arm_pats_diverge: Vec<_> = arms.iter().map(|arm| {
+        let all_arm_pats_diverge = arms.iter().map(|arm| {
```

Comment:
I think this change injected #55810, because it delayed our type-checking of the patterns. (Note the comment right above this line.)

Reply:
Needs a stronger comment, heh.

Reply:
Yeah I'll be fixing that.